src/protocols/output_management.rs

```diff
@@ -640,25 +643,15 @@ where
                         }
                     };
 
-                let Some(current_config) = g_state.current_state.get(output_id) else {
-                    warn!("SetMode: output missing from the current config");
-                    return;
-                };
-
-                let Some(mode) = current_config.modes.iter().find(|m| {
-                    m.width == width
-                        && m.height == height
-                        && (refresh == 0 || m.refresh_rate == refresh)
-                }) else {
-                    warn!("SetCustomMode: no matching mode");
-                    return;
-                };
-
-                new_config.mode = Some(niri_ipc::ConfiguredMode {
-                    width: mode.width,
-                    height: mode.height,
-                    refresh: Some(mode.refresh_rate as f64 / 1000.),
+                new_config.mode = Some(niri_config::output::Mode {
+                    custom: true,
+                    mode: niri_ipc::ConfiguredMode {
+                        width,
+                        height,
+                        refresh: Some(refresh as f64 / 1000.),
```

Comment:
The protocol [says](https://wayland.app/protocols/wlr-output-management-unstable-v1#zwlr_output_configuration_head_v1:request:set_custom_mode):

> If set to zero, the refresh rate is unspecified.

What do we do with it? I made it do nothing if refresh is 0 for now.